packages/aws-lambda/layer/bin/publish-layer.sh

```diff
@@ -459,7 +459,12 @@ echo "step 9/9: cleaning up"
 rm -rf $TMP_ZIP_DIR
 rm -rf $ZIP_NAME
 
-if [[ $BUILD_X86_64_FAIL -eq 1 && $BUILD_arm64_FAIL -eq 1 ]]; then
+echo "Build status: ARM64=$BUILD_arm64_FAIL, X86_64=$BUILD_X86_64_FAIL"
+
+# Check ARM64 build failure status only after all tasks have completed, 
+# since ARM64 is built after X86_64 and should not cause an early exit.
```

Comment:
# We only runt the condition below when the ARM64 build is running to avoid an early exit.